### PR TITLE
Fail deprecated initialization actions on Dataproc 2.0+

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+--------------------------------------------------------------------------------
+
+# NOTE: *The Docker initialization action has been deprecated. Please use the Docker Component*
+
+**The
+[Docker Component](https://cloud.google.com/dataproc/docs/concepts/components/docker)
```
Comment (functicons) [inline docker/README.md:6]: The link returns 404. Should we wait until it is available?

File: docker/docker.sh [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 set -euxo pipefail
 
+readonly NOT_SUPPORTED_MESSAGE="Docker initialization action is not supported on Dataproc 2.0+.
+Use Docker Component instead: https://cloud.google.com/dataproc/docs/concepts/components/docker"
+[[ $DATAPROC_VERSION = 2.* ]] && echo "$NOT_SUPPORTED_MESSAGE" && exit 1
```
Comment (functicons) [inline docker/docker.sh:7]: `!= 1.*` might be more future-proof.

File: docker/docker.sh [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 set -euxo pipefail
 
+readonly NOT_SUPPORTED_MESSAGE="Docker initialization action is not supported on Dataproc 2.0+.
```
Comment (functicons) [inline docker/docker.sh:5]: It is more accurate to replace 2.0+ with `${DATAPROC_VERSION}`.

File: flink/test_flink.py [hunk 4]
```diff
@@ -29,6 +30,10 @@ def __run_test_file(self, name, yarn_session):
       ("HA", ["m-0", "m-1", "m-2"]),
   )
   def test_flink(self, configuration, machine_suffixes):
+    # Skip on 2.0+ version of Dataproc because it's not supported
+    if self.getImageVersion() >= pkg_resources.parse_version("2.0"):
+      return
```
Comment (functicons) [inline flink/test_flink.py:35]: Is there a way to mark the test as skipped?

**Review by functicons — CHANGES_REQUESTED** [review]

File: docker/docker.sh (hunk 3 above)
Comment (medb) [reply docker/docker.sh]: Done.

File: flink/test_flink.py (hunk 4 above)
Comment (medb) [reply flink/test_flink.py]: Done.

File: docker/docker.sh (hunk 2 above)
Comment (medb) [reply docker/docker.sh]: Done.

**Review by functicons — APPROVED** [review]

File: docker/README.md (hunk 1 above)
Comment (medb) [reply docker/README.md:6]: Links already available.

**Pull request merged**